### PR TITLE
feat(websocket): communication server with client 

### PR DESCRIPTION
**Pull request by giovaz94**
Implmenting a webserver that enable a user to connect to the chat service and exchange messages with it

### CODE REVIEW

**Comment by giovaz94** [comment]
@tassiluca, I hope you'll find this usefull for working with websockets.
Note that the code is still on draft because it needs some minor adjustment and, obviously, documentation.

File: domain/src/main/scala/io/github/positionpal/message/ChatMessageADT.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  type ChatMessage = ChatMessageImpl
+
+  /**
+   * Represents the data that a ChatMessage exposes
+   */
+  trait MessageOps:
+    def text: String
+    def timestamp: String
+    def from: String            // TODO: Change return type to User
+    def to: String              // TODO: Change return type to Group
```
Comment (tassiluca) [inline domain/src/main/scala/io/github/positionpal/message/ChatMessageADT.scala:15]: todos comments, I don't know whether they are intentional or have been forgotten

**Review by tassiluca — APPROVED** [review]
Before merging can you please add the QA gradle plugin? I see it was removed [here](https://github.com/position-pal/chat-service/commit/00031241365e8f6c38ce0b57be5f10fd07bf94d0).
If you want to change the configurations you can place them in the root project directory named `.scalafix.conf` / `.scalafmt.conf`.

File: domain/src/main/scala/io/github/positionpal/message/ChatMessageADT.scala (hunk 1 above)
Comment (giovaz94) [reply domain/src/main/scala/io/github/positionpal/message/ChatMessageADT.scala:15]: Are intentional, the actual domain object will be added in the next feature release.

Comment (giovaz94) [reply domain/src/main/scala/io/github/positionpal/message/ChatMessageADT.scala]: Thank you for the catch on the QA plugin, It was removed by mistake.

File: domain/src/main/scala/io/github/positionpal/message/ChatMessageADT.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+object ChatMessageADT:
+
+  type ChatMessage = ChatMessageImpl
+
+  /** Represents the data that a ChatMessage exposes */
+  trait MessageOps:
+    def text: String
+    def timestamp: String
+    def from: String // TODO: Change return type to User
```
Comment (v4l3rio) [inline domain/src/main/scala/io/github/positionpal/message/ChatMessageADT.scala:11]: There is still a todo here, have you forgotten?

**Review by v4l3rio — APPROVED** [review]

**Pull request merged**